pkg/agent/attestor/node/node.go

```diff
@@ -93,6 +94,12 @@ func (a *attestor) loadSVID(ctx context.Context) ([]*x509.Certificate, *ecdsa.Pr
 		a.c.Log.Warn("Private key recovered, but no SVID found")
 	}
 
+	// If the retrieved SVID is expired, we should generate a new one.
+	if isSVIDValid(svid, time.Now) {
+		a.c.Log.Warn("Private key recovered, but SVID is expired.")
```

Comment:
I think this warning will fire if there is no private key, but there is a cached SVID? (e.g. if I am using keymanager memory?) We check if the private key was recovered just below. It might also fire if we got a private key but SVID wasn't found at all?

Errors at this stage are common during misconfiguration etc so I think it will be pretty important experience-wise to provide accurate error detection and messaging. I am wondering if we need to combine the above conditional, as well as the one below, into a switch statement or something that allows us to detect the combinations of error modes and handle it + log something that makes sense for each one?

Reply:
Agree this could use some cleanup, I have some ideas